bitcoin/src/sighash.rs

```diff
@@ -963,15 +963,9 @@ impl<R: Deref<Target = Transaction>> SighashCache<R> {
         self.segwit_cache.get_or_insert_with(|| {
             let common_cache = Self::common_cache_minimal_borrow(common_cache, tx);
             SegwitCache {
-                prevouts: sha256d::Hash::from_inner(
-                    sha256::Hash::hash(&common_cache.prevouts).into_inner(),
-                ),
-                sequences: sha256d::Hash::from_inner(
-                    sha256::Hash::hash(&common_cache.sequences).into_inner(),
-                ),
-                outputs: sha256d::Hash::from_inner(
-                    sha256::Hash::hash(&common_cache.outputs).into_inner(),
-                ),
+                prevouts: common_cache.prevouts.hash_again(),
+                sequences: common_cache.sequences.hash_again(),
+                outputs: common_cache.outputs.hash_again(),
```

Comment:
Hah, good memory! I totally forgot we added this `hash_again` method.

Reply:
Heh, I stared at it confused for a few minutes thinking "is this doing the same thing and we forgot to change it or does it do something else?!"

Reply:
I think we added it specifically for this code, and then forgot to use it.

Reply:
Yeah, I think the big advantage of having the crates in the same repo is we can use new APIs like this immediately which should prevent us from forgetting.